python/pyarrow/array.pxi

```diff
@@ -895,7 +895,7 @@ cdef class Array(_PandasConvertible):
             result = self.ap.Diff(deref(other.ap))
         return frombytes(result, safe=True)
 
-    def cast(self, object target_type, safe=True):
+    def cast(self, object target_type, safe=True, options=None):
```

Comment:
The redundancy between `safe` and `options` isn't great. I wonder if we can have `safe=None`, and then raise an exception if passing both `safe` and `options`?

Reply:
Yeah, exactly - we shouldn't allow passing both ideally since they're somewhat redundant